RMET-3975 - Setup Actions to run unit tests and release the library (#1)
ionic-team/ion-android-geolocation: docs/README.md

# OSGeolocationLib

The `OSGeolocationLib-Android` is a library built using `Kotlin` that provides geolocation features for Android applications.

The `OSGLOCController` class provides the main features of the Library, which are:
- obtaining the location/position of the device a single time;
- adding a watch to obtain periodic location updates;
- clearing/removing a previously added watch, turning off location updates.

## Index

- [Motivation](#motivation)
- [Usage](#usage)
- [Methods](#methods)
    - [Obtain the current location of the device](#obtain-the-current-location-of-the-device)
    - [Add a watch for periodic location updates](#add-a-watch-for-periodic-location-updates)
    - [Clear a watch that was added previously](#clear-a-watch-that-was-added-previously)

## Motivation

> This library is to be used by the Geolocation Plugin for [OutSystems' Cordova Plugin](https://github.com/ionic-team/cordova-outsystems-geolocation) and [Ionic's Capacitor Plugin](https://github.com/ionic-team/outsystems-geolocation).

Reviewer: Are you saying "to be used" as in it will be used in the future? While that is true now, it is going to be used very soon, so I think we don't need the "to be used" here.

```suggestion
This library is by [OutSystems' Cordova Geolocation Plugin](https://github.com/ionic-team/cordova-outsystems-geolocation) and [Ionic's Capacitor Geolocation Plugin](https://github.com/ionic-team/outsystems-geolocation).
```

Unless the phrase had a different meaning than I thought.

Author: I think we're reading the sentence in different ways 😆 . I believe your suggestion is missing a "used" though, so I'll commit a hybrid between our versions. Can you check?

Reviewer: Oh, yeah you're right, my suggestion is lacking too xD What you committed looks good to me :D

suspend fun getCurrentPosition(
    activity: Activity, 
    options: OSGLOCLocationOptions
): Result<OSGLOCLocationResult>
```

The method is composed of the following input parameters:
- **activity**: the `Activity` from the app using the library to use when obtaining the location.
- **options**: `OSGLOCLocationOptions` with the options with which to make the location request with (e.g. timeout).

The method returns a `Result` containing either an object of type `OSGLOCLocationResult`, which includes the geolocation data (e.g. latitide, longitude), or an exception that should be handled by the caller app.

### Add a watch for periodic location updates

```kotlin
fun addWatch(
    activity: Activity,
    options: OSGLOCLocationOptions,
    watchId: String
): Flow<Result<List<OSGLOCLocationResult>>>
```

The method is composed of the following input parameters:
- **activity**: the `Activity` from the app using the library to use when obtaining the location updates.
- **options**: `OSGLOCLocationOptions` with the options with which to make the location updates request with (e.g. timeout).
- **watchId**: a unique id identifying the watch to add, so that it can be removed later.

> The method returns a Flow in which hthe location updates will be emitted into.

Reviewer: ```suggestion
The method returns a Flow in which the location updates will be emitted to.
```
